### PR TITLE
chore(DEW): move to service folder and fix linter issues

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST='./huaweicloud/services/acceptance/dew' TESTARGS='-run=TestAccKms'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run=TestAccKms -timeout 360m -parallel 4
=== RUN   TestAccKmsKeyDataSource_Basic
=== PAUSE TestAccKmsKeyDataSource_Basic
=== RUN   TestAccKmsKeyDataSource_WithTags
=== PAUSE TestAccKmsKeyDataSource_WithTags
=== RUN   TestAccKmsKeyDataSource_WithEpsId
=== PAUSE TestAccKmsKeyDataSource_WithEpsId
=== RUN   TestAccKmsKey_Basic
--- PASS: TestAccKmsKey_Basic (33.82s)
=== RUN   TestAccKmsKey_Enable
=== PAUSE TestAccKmsKey_Enable
=== RUN   TestAccKmsKey_WithTags
=== PAUSE TestAccKmsKey_WithTags
=== RUN   TestAccKmsKey_WithEpsId
=== PAUSE TestAccKmsKey_WithEpsId
=== RUN   TestAccKmsKey_rotation
=== PAUSE TestAccKmsKey_rotation
=== CONT  TestAccKmsKeyDataSource_Basic
=== CONT  TestAccKmsKey_WithTags
=== CONT  TestAccKmsKey_rotation
=== CONT  TestAccKmsKey_WithEpsId
--- PASS: TestAccKmsKey_WithEpsId (20.54s)
=== CONT  TestAccKmsKeyDataSource_WithEpsId
--- PASS: TestAccKmsKey_WithTags (21.19s)
=== CONT  TestAccKmsKey_Enable
--- PASS: TestAccKmsKeyDataSource_Basic (22.71s)
=== CONT  TestAccKmsKeyDataSource_WithTags
--- PASS: TestAccKmsKey_rotation (41.61s)
--- PASS: TestAccKmsKeyDataSource_WithEpsId (22.67s)
--- PASS: TestAccKmsKeyDataSource_WithTags (27.55s)
--- PASS: TestAccKmsKey_Enable (36.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       91.678s
```

### CODE REVIEW

File: huaweicloud/services/dew/resource_huaweicloud_kms_key.go [hunk 1]
```diff
@@ -369,29 +392,31 @@ func resourceKmsKeyV1Update(d *schema.ResourceData, meta interface{}) error {
...
 	v, err := keys.Get(kmsKeyV1Client, d.Id()).ExtractKeyInfo()
 	if err != nil {
-		return CheckDeleted(d, err, "failed to retrieve key")
+		return common.CheckDeletedDiag(d, err, "failed to retrieve key")
 	}
 
 	deleteOpts := &keys.DeleteOpts{
-		KeyID: d.Id(),
+		KeyID:       d.Id(),
+		PendingDays: "7",
```
Comment (ShiChangkuo) [inline huaweicloud/services/dew/resource_huaweicloud_kms_key.go:415]: it seems that we can use `d.Get("pending_days").(string)` directly.

File: huaweicloud/services/dew/resource_huaweicloud_kms_key.go [hunk 2]
```diff
@@ -243,7 +246,7 @@ func resourceKmsKeyV1Read(d *schema.ResourceData, meta interface{}) error {
 	d.Set("key_id", v.KeyID)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dew/resource_huaweicloud_kms_key.go]: it's better to warp the error with multi-error

File: huaweicloud/provider.go [hunk 3]
```diff
@@ -791,7 +791,7 @@ func Provider() *schema.Provider {
 			"huaweicloud_iotda_device_certificate":  iotda.ResourceDeviceCertificate(),
 			"huaweicloud_iotda_device_linkage_rule": iotda.ResourceDeviceLinkageRule(),
 
-			"huaweicloud_kms_key":     ResourceKmsKeyV1(),
+			"huaweicloud_kms_key":     dew.ResourceKmsKeyV1(),
```
Comment (ShiChangkuo) [inline huaweicloud/provider.go]: the function name should be without version suffix

File: huaweicloud/services/dew/resource_huaweicloud_kms_key.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/dew/resource_huaweicloud_kms_key.go:415]: pending_days just been used by delete function, so remove the default value from schema.

File: huaweicloud/services/dew/resource_huaweicloud_kms_key.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/dew/resource_huaweicloud_kms_key.go]: done

File: huaweicloud/provider.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/provider.go]: ok,done

**Pull request merged**